cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -457,8 +455,15 @@ func (c *Client) addMember(name string, peerAddrs string, isLearner bool, promot
 
 func memberPromoteWithRety(ctx context.Context, cli *clientv3.Client, learnerID uint64) error {
 	klog.V(1).Infof("[etcd] Promoting a learner as a voting member: %016x", learnerID)
+	// The interval and timeout are following how talos does. See:
+	// https://github.com/siderolabs/talos/blob/474604cd279def7a6798e24ede27feef955ba5a3/internal/app/machined/pkg/system/services/etcd.go#L598-L599
 	err := wait.PollImmediate(15*time.Second, 10*time.Minute, func() (bool, error) {
-		// TODO: warning logs from etcd client should be removed
+		// TODO: warning logs from etcd client should be removed.
```

Comment:
iiuc, this TODO is for etcd client instead of kubeadm or k8s. I am okay to leave a TODO here but I suggest not comment something what we cannot do, it's a little confusing. e.g. someone reviews this comment some times later, is this something we can fixed by ourselves?

Reply:
As comments below, I think this should be handled once etcd takes a step forward. 
- We may pend on them, but this is still a to-do for us.